addons/sale_project/views/project_task_views.xml

```diff
@@ -18,6 +18,14 @@
         <field name="model">project.project</field>
         <field name="inherit_id" ref="project.edit_project"/>
         <field name="arch" type="xml">
+            <div class="oe_button_box" position="inside">
```

Comment:
Perhaps try ```
                <button class="d-none d-md-inline oe_stat_button"
                        type="object" name="action_view_so" icon="fa-dollar"
                        attrs="{'invisible': [('sale_order_id', '=', False)]}"
                        groups="sales_team.group_sale_salesman">
                       <div class="o_stat_info">
                          <field name="sale_order_id" attrs="{'invisible': True}"/>
                          <span class="o_stat_text">Sales Order</span>
                       </div>
              </button>```

Inspired from https://github.com/odoo/odoo/blob/14.0/addons/hr_holidays/views/hr_leave_type_views.xml#L25

Reply:
I just pushed the modification (the button still needs the be put in a div, otherwise an error occurs, and I removed the div inside the button: from my local tests, it is not needed)